identity/src/keypair.rs

```diff
@@ -224,42 +218,73 @@ impl Keypair {
                 Data: data.encode().to_vec(),
             },
             #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
-            Self::Rsa(_) => return Err(DecodingError::encoding_unsupported("RSA")),
+            Self::Rsa(data) => proto::PrivateKey {
+                Type: proto::KeyType::RSA,
+                Data: data.to_raw_bytes(),
+            },
             #[cfg(feature = "secp256k1")]
-            Self::Secp256k1(_) => return Err(DecodingError::encoding_unsupported("secp256k1")),
+            Self::Secp256k1(data) => proto::PrivateKey {
+                Type: proto::KeyType::Secp256k1,
+                Data: data.secret().to_bytes().into(),
+            },
             #[cfg(feature = "ecdsa")]
-            Self::Ecdsa(_) => return Err(DecodingError::encoding_unsupported("ECDSA")),
+            Self::Ecdsa(data) => proto::PrivateKey {
+                Type: proto::KeyType::ECDSA,
+                Data: data.secret().to_bytes(),
+            },
         };
 
         let mut buf = Vec::with_capacity(pk.get_size());
         let mut writer = Writer::new(&mut buf);
         pk.write_message(&mut writer).expect("Encoding to succeed");
 
-        Ok(buf)
+        buf
     }
 
     /// Decode a private key from a protobuf structure and parse it as a [`Keypair`].
-    #[cfg_attr(not(feature = "ed25519"), allow(unused_mut))]
+    #[deprecated(
+        since = "0.2.0",
+        note = "This method name is inaccurate, use `Keypair::try_from_protobuf_encoding` instead."
+    )]
     pub fn from_protobuf_encoding(bytes: &[u8]) -> Result<Keypair, DecodingError> {
+        Self::try_from_protobuf_encoding(bytes)
+    }
+
+    /// Try to decode a private key from a protobuf structure and parse it as a [`Keypair`].
+    #[cfg_attr(not(feature = "ed25519"), allow(unused_mut))]
+    pub fn try_from_protobuf_encoding(bytes: &[u8]) -> Result<Keypair, DecodingError> {
```

Comment:
This could be `try_decode_from_protobuf` then, based on the naming suggestion for the other one.